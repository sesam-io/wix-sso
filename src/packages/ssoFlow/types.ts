import type { Auth0ClientOptions } from "@auth0/auth0-spa-js";

export enum LoginMessageType {
  Login = "auth0:login",
  Logout = "auth0:logout",
  Signup = "auth0:signup",
}

export type RunSSOFlowArgs = {
  auth0ClientOptions: Auth0ClientOptions;
  siteId?: string | "";
};

export type LoadScriptArgs = {
  name: string;
  idAttribute?: string;
  url: string;
};

// The datalayer object is set on the window object by the Google Analytics script
// it can be used to configure GA, set options or send specific events
declare global {
  interface Window {
    dataLayer: unknown[];
  }
}
