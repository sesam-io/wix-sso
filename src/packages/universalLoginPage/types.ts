import { SiteIds } from "./constants";

export type FormType = "login" | "logout";

export type SiteId = keyof typeof SiteIds;

export type Sites = {
  [key: string]: {
    title: string;
    logoUrl?: string;
    logIn: {
      subTitle: string;
    };
    signup: {
      subTitle: string;
    };
  };
};
