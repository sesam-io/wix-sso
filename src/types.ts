export enum LoginMessageType {
  Login = "auth0:login",
  Logout = "auth0:logout",
}

export type LoadScriptArgs = {
  name: string;
  idAttribute?: string;
  url: string;
};
