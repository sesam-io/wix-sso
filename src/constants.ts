export const Auth0_SPA_JS_CDN =
  "https://cdn.auth0.com/js/auth0-spa-js/2.0/auth0-spa-js.production.js";

export const ZENDESK_WIDGET_CDN =
  "https://static.zdassets.com/ekr/snippet.js?key=eb7f5552-be33-4b0f-a55d-ce9a8a7aa975";
