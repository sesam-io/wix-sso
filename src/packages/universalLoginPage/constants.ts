import type { Sites } from "./types";

export const LOGO_IMG_ID = "prompt-logo-center";

export const SiteIds = {
  hubspot: "HubSpot",
  poweroffice: "PowerOffice",
  sesam: "Sesam",
  superoffice: "SuperOffice",
  tripletex: "Tripletex",
  wave: "Wave",
};

const SITE_LOGOS_RELEASE_VERSION = "v1.0.94-site-logos";

export const BASE_LOGO_URL = `https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@${SITE_LOGOS_RELEASE_VERSION}/src/packages/siteLogos`;

const DEFAULT_LOGO_URL = `${BASE_LOGO_URL}/sesam-talk-rgb.png`;

export const WixSites: Sites = {
  hubspot: {
    id: "hubspot",
    logoUrl: `${BASE_LOGO_URL}/making-hubspot-talk-logo-centered.svg`,
    loginSubTitle: "Log in to Making HubSpot Talk.",
    signupSubTitle: "Sign up to Making HubSpot Talk.",
  },
  poweroffice: {
    id: "poweroffice",
    logoUrl: `${BASE_LOGO_URL}/powerofficego/powerofficego-mc-header-logo.svg`,
    loginSubTitle: "Log in to PowerOffice Data Sync.",
    signupSubTitle: "Sign up to PowerOffice Data Sync.",
  },
  sesam: {
    id: "sesam",
    logoUrl: DEFAULT_LOGO_URL,
    loginSubTitle: "Log in to Sesam Talk.",
    signupSubTitle: "Sign up to Sesam Talk to continue to Sesam Talk.",
  },
  superoffice: {
    id: "superoffice",
    logoUrl: `${BASE_LOGO_URL}/superoffice-mc-header-logo.svg`,
    loginSubTitle: "Log in to SuperOffice Data Sync.",
    signupSubTitle: "Sign up to SuperOffice Data Sync.",
    titleClassName: "superofficeLogInTitle",
  },
  tripletex: {
    id: "tripletex",
    logoUrl: DEFAULT_LOGO_URL,
    loginSubTitle: "Log in to Making Tripletex Talk.",
    signupSubTitle: "Sign up to Making Tripletex Talk.",
  },
  wave: {
    id: "wave",
    logoUrl: `${BASE_LOGO_URL}/making-wave-talk-logo-centered.svg`,
    loginSubTitle: "Log in to Making Wave Talk.",
    signupSubTitle: "Sign up to Making Wave Talk.",
  },
};
