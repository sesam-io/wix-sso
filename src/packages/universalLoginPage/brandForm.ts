import { BASE_LOGO_URL, WixSites } from "./constants";
import { FormType, SiteId } from "./types";

export const getWixSite = (siteId: SiteId) => {
  const site = WixSites[siteId];

  if (!site) {
    return WixSites.sesam;
  }

  return site;
};

export const getDefaultPageTitle = (formType: FormType) =>
  formType === "login"
    ? WixSites.sesam.loginSubTitle
    : WixSites.sesam.signupSubTitle;

export const brandLogo = (imgElement: HTMLImageElement, logoUrl?: string) => {
  if (logoUrl) {
    imgElement.src = logoUrl;
  }
};

export const getBrandTitleFn =
  (pTags: HTMLCollectionOf<HTMLParagraphElement>, defaultPageTitle: string) =>
  (subTitle: string, titleClassName?: string) => {
    Array.from(pTags)?.forEach((pTag) => {
      if (pTag.textContent?.toLowerCase() === defaultPageTitle.toLowerCase()) {
        pTag.innerText = subTitle;

        if (titleClassName) {
          pTag.className = titleClassName;
        }
      }
    });
  };

export const insertElementAfter = (
  referenceNode: HTMLImageElement,
  newNode: HTMLDivElement
) => {
  referenceNode?.parentNode?.insertBefore(newNode, referenceNode.nextSibling);
};

export const addPoweredBySesamImg = (imgElement: HTMLImageElement) => {
  const poweredBySesamWrapper = document.createElement("div");
  const poweredBySesamImg = document.createElement("img");

  poweredBySesamImg.src = `${BASE_LOGO_URL}/sesam/powered-by-sesam.svg`;
  poweredBySesamImg.className = "poweredBySesamImg";
  poweredBySesamWrapper.appendChild(poweredBySesamImg);
  poweredBySesamWrapper.className = "poweredBySesamWrapper";

  insertElementAfter(imgElement, poweredBySesamWrapper);
};

export const buildBrandedHorizontalLogo = async (
  imgElement: HTMLImageElement,
  htmlLogoUrl: string
) => {
  const poweredBySesamWrapper = document.createElement("div");

  fetch(htmlLogoUrl)
    .then((response) => response.text())
    .then((text) => {
      poweredBySesamWrapper.innerHTML = text;
      poweredBySesamWrapper.style.display = "flex";
      poweredBySesamWrapper.style.justifyContent = "center";
    });

  insertElementAfter(imgElement, poweredBySesamWrapper);
  imgElement.remove();
};
