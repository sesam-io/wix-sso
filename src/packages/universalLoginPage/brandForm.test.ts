/**
 * @jest-environment jsdom
 */

import { getBrandTitleFn, getWixSite, getDefaultPageTitle } from "./brandForm";
import { WixSites } from "./constants";
import { SiteId } from "./types";

describe("getWixSite test suite", () => {
  it("should return default site", () => {
    expect(getWixSite("none-exist-site" as SiteId)).toMatchSnapshot();
  });

  it("should return sesam site", () => {
    expect(getWixSite("sesam")).toMatchSnapshot();
  });

  it("should return Wave site", () => {
    expect(getWixSite("wave")).toMatchSnapshot();
  });

  it("should return SuperOffice site", () => {
    expect(getWixSite("superoffice")).toMatchSnapshot();
  });
});

describe("brandTitle test suite", () => {
  it("should brand login Wave site subtitle", () => {
    document.body.innerHTML = `<div>
        <p>Some title</p>
        <p>${WixSites.sesam.loginSubTitle}</p>
        <p>Some text</p>
    </div>`;

    const pTags = document.getElementsByTagName("p");

    const brandTitle = getBrandTitleFn(pTags, getDefaultPageTitle("login"));

    brandTitle(WixSites.wave.loginSubTitle);
  });
});
