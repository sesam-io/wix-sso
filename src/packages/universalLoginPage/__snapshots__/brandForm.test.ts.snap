// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getWixSite test suite should return SuperOffice site 1`] = `
{
  "id": "superoffice",
  "loginSubTitle": "Log in to SuperOffice Data Sync.",
  "logoUrl": "https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@v1.0.119-site-logos/src/packages/siteLogos/superoffice/superoffice-ulp-header-logo.svg",
  "signupSubTitle": "Sign up to SuperOffice Data Sync.",
  "titleClassName": "superofficeLogInTitle",
}
`;

exports[`getWixSite test suite should return Wave site 1`] = `
{
  "id": "wave",
  "loginSubTitle": "Log in to Making Wave Talk.",
  "logoUrl": "https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@v1.0.119-site-logos/src/packages/siteLogos/making-wave-talk-logo-centered.svg",
  "signupSubTitle": "Sign up to Making Wave Talk.",
}
`;

exports[`getWixSite test suite should return default site 1`] = `
{
  "id": "sesam",
  "loginSubTitle": "Log in to Sesam Talk.",
  "logoUrl": "https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@v1.0.119-site-logos/src/packages/siteLogos/sesam-talk-rgb.png",
  "signupSubTitle": "Sign up to Sesam Talk to continue to Sesam Talk.",
}
`;

exports[`getWixSite test suite should return sesam site 1`] = `
{
  "id": "sesam",
  "loginSubTitle": "Log in to Sesam Talk.",
  "logoUrl": "https://cdn.jsdelivr.net/gh/sesam-io/wix-sso@v1.0.119-site-logos/src/packages/siteLogos/sesam-talk-rgb.png",
  "signupSubTitle": "Sign up to Sesam Talk to continue to Sesam Talk.",
}
`;
